### PR TITLE
Add auth-capture flow and tests

### PR DESCRIPTION
**Pull request by daliwei**
@pierre @sbrossie Please review.

### CODE REVIEW

File: lib/paypal_express/api.rb [hunk 1]
```diff
@@ -29,20 +29,72 @@ def on_event(event)
       end
 
       def authorize_payment(kb_account_id, kb_payment_id, kb_payment_transaction_id, kb_payment_method_id, amount, currency, properties, context)
```
Comment (pierre) [inline lib/paypal_express/api.rb]: Would it be possible to share the code (it's pretty much the same as `purchase_payment`)?

AFAICT, the main differences are just:

* `transaction_type` and `api_call` (in non HPP flow) when creating the response row
* The `ActiveMerchant` call in the gateway proc

**Comment by pierre** [comment]
Looking good overall, just a few small comments!

File: lib/paypal_express/api.rb [hunk 2]
```diff
@@ -266,16 +320,32 @@ def build_form_descriptor(kb_account_id, descriptor_fields, properties, context)
...
-                                             kb_payment_method.id,
-                                             nil,
-                                             amount,
-                                             currency,
-                                             payment_external_key,
-                                             transaction_external_key,
-                                             custom_props,
-                                             jcontext)
+
+          auth_mode = ::Killbill::Plugin::ActiveMerchant::Utils.normalized(options, :auth_mode)
```
Comment (pierre) [inline lib/paypal_express/api.rb]: Could you document the new property in the `README`?

**Comment by daliwei** [comment]
Hey, @pierre , updated based on your comment. Please take a look 😄

**Comment by pierre** [comment]
Awesome!

**Comment by sbrossie** [comment]
@daliwei Good job on the tests too!

**Pull request merged**